org.eclipse.jdt.ls.product/scripts/jdtls.py

```diff
@@ -30,7 +30,7 @@ def get_java_executable(validate_java_version):
 
 	out = subprocess.check_output([java_executable, '-version'], stderr = subprocess.STDOUT, universal_newlines=True)
 
-	matches = re.finditer(r"(?P<major>\d+)\.\d+\.\d+", out)
+	matches = re.finditer(r"(?<=version\s\")(?P<major>\d+)(\.\d+\.\d+(_\d+)?)?", out)
```

Comment:
Is there any use in matching `(_\d+)?` (the part after the minor & security number) ? Would it not still match ?

Overall, looks fine to me and still handles the cases of :

`openjdk version "11.0.14.1" 2022-02-08`
`openjdk version "18" 2022-03-22`

```

Reply:
Some old jdk versions had version schema like `1.8.0_322`, I'm not sure if this is still relevant.
The above match is optional so it does not hurt having it in the regex.

Reply:
This doesn't work with the Early Access of openjdk: `openjdk version "22-ea" 2024-03-19`

Reply:
Nevermind, it seems since the last group is optional, it still matches, but this is something to consider in the future